compiler/ir/backend.wasm/src/org/jetbrains/kotlin/backend/wasm/lower/WasmTypeOperatorLowering.kt

```diff
@@ -184,6 +184,7 @@ class WasmBaseTypeOperatorTransformer(val context: WasmBackendContext) : IrEleme
 
         // Handled by autoboxing transformer
         if (toType.isInlined() && !fromType.isInlined()) {
+            if (fromType.isUnit()) return value
```

Comment:
This was making multiple tests fail in `WasmFirWasmInvalidationPerFileTestGenerated`, by constructing a code like this:
```
i32.const 1
call $kotlin.wasm.internal.<set-isNotFirstWasmExportCall>___fun_1534
global.get $kotlin.Unit_instance___g_513
ref.cast $kotlin.Boolean___type_432
struct.get_s $kotlin.Boolean___type_432 4  ;; name: value, type: kotlin.Boolean
drop
```

The cause of these failures was an illegal cast problem. This problem happens because the call to the setter of `isNotFirstWasmExportCall` (and maybe others), which returns `Unit`, was wrapped by an unboxing intrinsic function call, casting the resulting Unit to a Boolean, then accessing its i8 field.

Reply:
Well, so we need to fix it but not workaround.

Reply:
I get that you mean to fix the source boxing the `Unit`. I just don't understand, what does it mean to box a `Unit` type into an `Int` type?

Reply:
Ok, let say - why this line needed? Why it works on `master` and does not works in our branch? There is some bug in the `master`? Can we then make a reproducer and file an issue there?

Reply:
Fixed in [d9ba0b6](https://github.com/JetBrains/kotlin/pull/5312/commits/d9ba0b61c2f2aafb1ac01fd9b3ef59e9d3d90e36)

Reply:
This is seems to be a bug in `master`. Looks weird why it was not a problem in master. Maybe you can write example which is fire in `master` and we can push this fix separately?

Reply:
Sure

Reply:
It's not showing in `master` because the `propertyAccessorInlinerLoweringPhase` optimization is not disabled (it must be disabled for now for IC to work correctly, as per described in the comment in [WasmLoweringPhases.kt](https://github.com/JetBrains/kotlin/pull/5312/files#diff-3db36f2996a3df822b3f9705b161132f831e52ac4e44b201234a2044a3f1e7d9L715)). The lowering turns the call to the setters to a `global.set`, hiding the bug in these specific setters.

We can't construct a failing test in this case because this is happening in a single exact place, namely in [UnhandledExceptionLowering.kt](https://github.com/JetBrains/kotlin/blob/master/compiler/ir/backend.wasm/src/org/jetbrains/kotlin/backend/wasm/lower/UnhandledExceptionLowering.kt#L86) (actually in [InvokeOnExportedFunctionExitLowering.kt](https://github.com/OsamaAhmad00/kotlin/blob/signatures-for-lowerings/compiler/ir/backend.wasm/src/org/jetbrains/kotlin/backend/wasm/lower/InvokeOnExportedFunctionExitLowering.kt#L87) too), and the bug is hidden by the fact that the erroneous setter is being inlined.

Both instances are fixed [dbc0fe5](https://github.com/JetBrains/kotlin/pull/5312/commits/dbc0fe505d6dacfb1bb23152300859e0e59f2bc1)

I'll fix them and open a PR to the `master` branch, but without a failing test in this case.